### PR TITLE
Test s2i build in disconnect cluster

### PR DESCRIPTION
**Pull request by xiuwang**
@openshift/devexp-qe Please help review this pr, here is log http://pastebin.test.redhat.com/850070.
Disconnect cluster can't access github.com or other external source, then setup a repo using gitserver, which will be used in s2i build.

### CODE REVIEW

**Comment by liangxia** [comment]
Failure from git clone,
``` 
STEP 4: RUN umask 0002 && /clonerefs && find /go/src -type d -not -perm -0775 | xargs -r chmod g+xw
{"component":"clonerefs","file":"prow/pod-utils/clone/clone.go:45","func":"k8s.io/test-infra/prow/pod-utils/clone.Run","level":"info","msg":"Cloning refs","refs":{"org":"openshift","repo":"verification-tests","repo_link":"https://github.com/openshift/verification-tests","base_ref":"master","base_sha":"64144452508e838b4c144c6de385f30bd09a67af","base_link":"https://github.com/openshift/verification-tests/commit/64144452508e838b4c144c6de385f30bd09a67af","pulls":[{"number":891,"author":"xiuwang","sha":"f3cf2cfdac8e1a98f33f02b7902b3dde1edf21e2","link":"https://github.com/openshift/verification-tests/pull/891","commit_link":"https://github.com/openshift/verification-tests/pull/891/commits/f3cf2cfdac8e1a98f33f02b7902b3dde1edf21e2","author_link":"https://github.com/xiuwang"}]},"time":"2020-04-01T02:48:29Z"}
{"command":"mkdir -p /go/src/github.com/openshift/verification-tests","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"","time":"2020-04-01T02:48:29Z"}
{"command":"git init","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"Initialized empty Git repository in /go/src/github.com/openshift/verification-tests/.git/\n","time":"2020-04-01T02:48:29Z"}
{"command":"git config user.name ci-robot","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"","time":"2020-04-01T02:48:29Z"}
{"command":"git config user.email ci-robot@openshift.io","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"","time":"2020-04-01T02:48:29Z"}
{"command":"git fetch https://github.com/openshift/verification-tests.git --tags --prune","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"","time":"2020-04-01T02:48:29Z"}
{"command":"git fetch https://github.com/openshift/verification-tests.git master","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"From https://github.com/openshift/verification-tests\n * branch            master     -\u003e FETCH_HEAD\n","time":"2020-04-01T02:48:30Z"}
{"command":"git checkout 64144452508e838b4c144c6de385f30bd09a67af","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"Note: checking out '64144452508e838b4c144c6de385f30bd09a67af'.\n\nYou are in 'detached HEAD' state. You can look around, make experimental\nchanges and commit them, and you can discard any commits you make in this\nstate without impacting any branches by performing another checkout.\n\nIf you want to create a new branch to retain commits you create, you may\ndo so (now or later) by using -b with the checkout command again. Example:\n\n  git checkout -b new_branch_name\n\nHEAD is now at 6414445... Fix occacional failure of ci job (#898)\n","time":"2020-04-01T02:48:30Z"}
{"command":"git branch --force master 64144452508e838b4c144c6de385f30bd09a67af","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"","time":"2020-04-01T02:48:30Z"}
{"command":"git checkout master","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"Switched to branch 'master'\n","time":"2020-04-01T02:48:30Z"}
{"command":"git fetch https://github.com/openshift/verification-tests.git pull/891/head","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"From https://github.com/openshift/verification-tests\n * branch            refs/pull/891/head -\u003e FETCH_HEAD\n","time":"2020-04-01T02:48:30Z"}
{"command":"git merge --no-ff f3cf2cfdac8e1a98f33f02b7902b3dde1edf21e2","component":"clonerefs","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"Merge made by the 'recursive' strategy.\n features/build/stibuild.feature                   | 27 +++++++++++++++++++++++\n testdata/build/httpd-ex                           |  1 +\n testdata/image/gitserver/gitserver-ephemeral.yaml |  5 +++--\n 3 files changed, 31 insertions(+), 2 deletions(-)\n create mode 160000 testdata/build/httpd-ex\n","time":"2020-04-01T02:48:31Z"}
{"command":"git submodule update --init --recursive","component":"clonerefs","error":"exit status 1","file":"prow/pod-utils/clone/clone.go:57","func":"k8s.io/test-infra/prow/pod-utils/clone.Run.func2","level":"info","msg":"Ran command","output":"No submodule mapping found in .gitmodules for path 'testdata/build/httpd-ex'\n","time":"2020-04-01T02:48:31Z"}
{"component":"clonerefs","error":"one or more of the records are in failed state","file":"prow/cmd/clonerefs/main.go:40","func":"main.main","level":"fatal","msg":"Failed to clone refs","time":"2020-04-01T02:48:31Z"}
subprocess exited with status 1
subprocess exited with status 1
error: build error: error building at STEP "RUN umask 0002 && /clonerefs && find /go/src -type d -not -perm -0775 | xargs -r chmod g+xw": exit status 1
2020/04/01 02:48:50 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/04/01 02:48:51 Ran for 1m13s
2020/04/01 02:48:51 Submitted failure event to sentry (id=67100aa80ec4488b8d3c694c08761420)
error: could not run steps: step src failed: could not wait for build: the build src failed after 59s with reason DockerBuildFailed: Dockerfile build strategy has failed.
{"command":"git submodule update --init --recursive","comp...'testdata/build/httpd-ex'\n","time":"2020-04-01T02:48:31Z"}
{"component":"clonerefs","error":"one or more of the recor..."msg":"Failed to clone refs","time":"2020-04-01T02:48:31Z"}
subprocess exited with status 1
subprocess exited with status 1
error: build error: error building at STEP "RUN umask 0002...pe d -not -perm -0775 | xargs -r chmod g+xw": exit status 1
```

/retest

**Comment by xiuwang** [comment]
@liangxia Could you help review, again? Thanks

File: features/build/stibuild.feature [hunk 1]
```diff
@@ -135,3 +135,26 @@ Feature: stibuild.feature
...
+    And I run the :set_env client command with:
+      | resource | dc/git               |
+      | e        | REQUIRE_SERVER_AUTH= |
+      | e        | REQUIRE_GIT_AUTH=    |
+    Then the step should succeed
+    When a pod becomes ready with labels:
+      | deploymentconfig=git |
+      | deployment=git-2     |
+    When I run the :cp client command with:
+      | source | <%= ENV['BUSHSLICER_HOME'] %>/testdata/build/httpd-ex.git |
```
Comment (liangxia) [inline features/build/stibuild.feature:153]: @xiuwang  Could you try to replace `ENV['BUSHSLICER_HOME']` with `BushSlicer::HOME`

Comment (wzheng1) [reply features/build/stibuild.feature:153]: @liangxia why? any difference?

Comment (liangxia) [reply features/build/stibuild.feature:153]: No differences. Just a little shorter, more readable, and with less special characters.

Comment (xiuwang) [reply features/build/stibuild.feature]: @liangxia Updated, and passed in my local host, please help review again. Thanks

**Comment by liangxia** [comment]
LGTM.
I'm not sure what those bin files are.  I'll let @pruan-rht to take a look, too.

**Comment by xiuwang** [comment]
> LGTM.
> I'm not sure what those bin files are. I'll let @pruan-rht to take a look, too.
testdata/build/httpd-ex.git is a git repo, the binary is genereted by cmd `git clone --bare <url> httpd-ex.git` , it records commit info, can't be removed(I have verified).

File: features/build/stibuild.feature [hunk 2]
```diff
@@ -135,3 +135,26 @@ Feature: stibuild.feature
...
+    Then the step should succeed
+    When a pod becomes ready with labels:
+      | deploymentconfig=git |
+      | deployment=git-2     |
+    When I run the :cp client command with:
+      | source | <%= BushSlicer::HOME %>/testdata/build/httpd-ex.git | 
+      | dest   | <%= pod.name %>:/var/lib/git/                       |
+    Then the step should succeed
+    When I run the :new_app client command with:
+      | app_repo  | openshift/httpd:latest~http://<%= cb.git_route %>/httpd-ex.git |
```
Comment (liangxia) [inline features/build/stibuild.feature:157]: Extra white spaces between `app_repo` and the second `|`

Comment (xiuwang) [reply features/build/stibuild.feature]: @liangxia Updated, thx

**Pull request merged**